server/pulp/server/db/model/criteria.py

```diff
@@ -499,9 +499,13 @@ def apply(query):
             if matched:
                 query[key] = translated
                 continue
-            if not isinstance(value, dict):
+            if isinstance(value, dict):
+                DateOperator.apply(value)
                 continue
-            DateOperator.apply(value)
+            if isinstance(value, list):
+                for item in value:
+                    if isinstance(item, dict):
+                        DateOperator.apply(item)
```

Comment:
It looks like it solves only one level of nesting.
More complex queries don't seem to be covered:
```
        query = {
￼            '$and': [
￼                {'created': {
￼                    '$gt': {'$date': '2013-12-01T14:35:00Z'}
￼                }},
                {'$or' : [
￼                    {'updated': {
￼                        '$lt': {'$date': '2020-05-15T14:35:00Z'}
￼                    }},
                    {'updated': {
￼                        '$gt': {'$date': '2020-05-20T00:00:00Z'}
￼                    }},
                ]},
￼            ]
￼        }
```

Reply:
I think this will work as it's a recursive call to DateOperator.apply() for each item in the list that will go to any level of nesting. Also, tried this example too and it works fine.

Reply:
It won't descend into lists of lists though.  If there is any list within a list, then we break out of recursion.  It seems slightly wrong but my guess is that Pulp never contains data where such a query makes sense anyway.

Reply:
Agree. It's with the assumption that it will either be dict or list of dict.